test/GroupByTest.cpp

```diff
@@ -421,7 +445,100 @@ TEST_F(GroupBySpecialCount, getPermutationForThreeVariableTriple) {
 }
 
 // _____________________________________________________________________________
-TEST_F(GroupBySpecialCount, checkIfJoinWithFullScan) {
+TEST_F(GroupByOptimizations, checkIfHashMapOptimizationPossible) {
+  auto testFailure = [this](const auto& groupByVariables, const auto& aliases,
+                            const auto& join, const auto& aggregates) {
+    auto groupBy = GroupBy{qec, groupByVariables, aliases, join};
+    ASSERT_FALSE(groupBy.checkIfHashMapOptimizationPossible(aggregates));
+  };
+
+  std::vector<Variable> variablesXAndY{varX, varY};
+
+  std::vector<ColumnIndex> sortedColumnsVector = {0};
+  Tree validJoinWhenGroupingByXPlusSort = makeExecutionTree<Sort>(
```

Comment:
Maybe just call this `subtreeWithSort` because you don't really care about the contents of the subsequence.